indico/modules/events/registration/logging.py

```diff
@@ -0,0 +1,56 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2020 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from __future__ import unicode_literals
+
+from flask import session
+
+from indico.core import signals
+from indico.modules.events import EventLogKind, EventLogRealm
+from indico.modules.events.registration.models.registrations import RegistrationState
+
+
+def connect_log_signals():
+    signals.event.registration_checkin_updated.connect(log_registration_check_in)
+    signals.event.registration_state_updated.connect(log_registration_updated)
+
+
+def log_registration_check_in(sender, *args, **kwargs):
+    """Log a registration check-in action to the event log."""
+    registration = sender
+    if registration.checked_in:
+        log_text = '"{}" has been checked in'.format(registration.full_name)
+    else:
+        log_text = '"{}" check-in has been reset'
+    registration.event.log(EventLogRealm.participants, EventLogKind.change, 'Event',
+                           log_text.format(registration.full_name), session.user)
+
+
+def log_registration_updated(sender, previous_state, *args, **kwargs):
+    """Log the registration status change to the event log."""
+    from indico.modules.events.registration import logger
+    registration = sender
+    if not previous_state:
+        return
+    status_descriptions = {
+        RegistrationState.complete: 'approved',
+        RegistrationState.rejected: 'rejected',
+    }
+    if registration.state == RegistrationState.complete:
+        log_text = 'Registration for "{}" has been approved'
+    elif registration.state == RegistrationState.rejected:
+        log_text = 'Registration for "{}" has been rejected'
+    elif registration.state == RegistrationState.pending:
+        old_state = status_descriptions.get(previous_state)
+        if not old_state:
+            old_state = str(old_state).split('.')[-1]
```

Comment:
that looks quite strange to me...

Reply:
it's a bit tricky but, the `__nonzero__` of a `TransactionStatus.pending` is evaluated to `False` while the `__str__` of it is: "TransactionStatus.pending", and `str.split('.')[-1] is "pending"

Reply:
Updated in favor of making use of `RegistrationState` titles.

```python
previous_state_title = orig_string(previous_state.title).lower()
log_text = 'Registration for "{}" has been changed from %s to pending' % previous_state_title
```